pkg/controller/main-controller.go

```diff
@@ -1118,49 +1115,47 @@ func (c *Controller) syncHandler(key string) error {
 
 		us, err := adminClnt.ServerUpdate(ctx, updateURL)
 		if err != nil {
-			_ = c.removeArtifacts()
-
-			err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", tenantName, err)
-			if _, terr := c.updateTenantStatus(ctx, tenant, err.Error(), totalAvailableReplicas); terr != nil {
-				return terr
-			}
-
-			// Update failed, nothing needs to be changed in the container
-			return err
-		}
+			if err.(madmin.ErrorResponse).Code != "MethodNotAllowed" {
+				err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", tenantName, err)
```

Comment:
```suggestion
				err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", tenantName, err)
```

Do not need to overwrite top level `err` with a custom message just to update tenantStatus() instead use `fmt.Sprintf()`